src/azul/service/hca_response_v5.py

```diff
@@ -386,28 +383,37 @@ def make_organoid(self, organoid):
     def make_organoids(self, entry):
         return [self.make_organoid(organoid) for organoid in entry["contents"]["organoids"]]
 
-    def make_sample(self, sample):
+    def make_sample(self, sample, entity_type):
         lookup = {
             'cell_lines': ('cellLines', self.make_cell_line),
             'organoids': ('organoids', self.make_organoid),
             'specimens': ('specimens', self.make_specimen),
         }
```

Comment:
Excellent refactoring suggestion. Think I'll go one more and just pass in the `entity_dict` into `make_sample()` instead of the `make_function` which is only needed so that the `entity_dict` can be created.